### PR TITLE
Add custom body to Request

### PR DESCRIPTION
**Pull request by ivan-iver**
When I need to send my post in different format (xml instead) always sent as json. Now you can send a custom body with a BodyReader field as io.Reader type.
Ej.:

```go
res, err := goreq.Request{
    Method: "POST",
    Uri: "http://www.google.com",
    ContentType: "application/x-www-form-urlencoded",
    BodyReader: strings.NewReader(params.Encode()),
}.Do()
```

### CODE REVIEW

**Comment by marcosnils** [comment]
@ivan-iver sorry for taking too long to review. Can you please add a test for this so we can merge?

File: goreq.go [hunk 1]
```diff
@@ -30,6 +30,7 @@ type Request struct {
 	Method            string
 	Uri               string
 	Body              interface{}
+	BodyReader        io.Reader
```
Comment (marcosnils) [inline goreq.go]: Why is this necessary if `prepareRequestBody` already returns a reader?. Can't we use that?

File: goreq.go [hunk 2]
```diff
@@ -436,7 +437,9 @@ func (r Request) NewRequest() (*http.Request, error) {
 	}
 
 	var bodyReader io.Reader
-	if b != nil && r.Compression != nil {
+	if r.BodyReader != nil {
```
Comment (marcosnils) [inline goreq.go]: Why is this necessary?. If `Body` is a reader then the `prepareRequestBody` will return a `Reader` which will be assigned here: https://github.com/franela/goreq/pull/126/files#diff-04563bc79ed43224b4d9ebd57a2ad279R456

Comment (ivan-iver) [inline goreq.go]: You are right. I didn't see previously. Was my mistake don't read all code. I can send my request with something like:

```
res, err := goreq.Request{
    Method: "POST",
    Uri: "http://www.google.com",
    ContentType: "application/x-www-form-urlencoded",
    Body: params.Encode(),
}.Do()
```
An apology.

**Comment by ivan-iver** [comment]
This pull request is not necessary.
